src/InsertionsClient.Console/ConsoleApp/BuildFilterFactory.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.DotNet.InsertionsClient.Models;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq.Expressions;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DotNet.InsertionsClient.ConsoleApp
+{
+    internal static class BuildFilterFactory
```

Comment:
minor: Builder & Factory are 2 distinct patterns, is it possible to update the class' name to best reflect the intended pattern?

Reply:
@donJoseLuis  Build here means "SDK after it was compiled". These builds are all listed in the manifest file. So this factory is instantiating filters that decides if a build should be inserted or not. Therefore, it is a "BuildFilter-Factory". Does this make a bit more sense or do you think we should avoid using the word "Build" here?

Reply:
I understood the meaning from the get go. If there is another type name that abstracts the same concept, we should go for that, when thinking about code maintenance.